pkg/resources/schema.go

```diff
@@ -29,268 +33,380 @@ var schemaSchema = map[string]*schema.Schema{
+		quotedIdentifiersIgnoreCase,
+		enableConsoleOutput,
+		err := GetAllDatabaseParameters(d)
+	if err != nil {
+		return diag.FromErr(err)
 	}
 
-	dataRetentionTimeInDays := GetPropertyAsPointer[int](d, "data_retention_days")
-	if dataRetentionTimeInDays != nil && *dataRetentionTimeInDays != IntDefault {
-		createReq.DataRetentionTimeInDays = dataRetentionTimeInDays
+	opts := &sdk.CreateSchemaOptions{
+		Transient:                               GetConfigPropertyAsPointerAllowingZeroValue[bool](d, "is_transient"),
+		WithManagedAccess:                       GetConfigPropertyAsPointerAllowingZeroValue[bool](d, "with_managed_access"),
+		DataRetentionTimeInDays:                 dataRetentionTimeInDays,
+		MaxDataExtensionTimeInDays:              maxDataExtensionTimeInDays,
+		ExternalVolume:                          externalVolume,
+		Catalog:                                 catalog,
+		ReplaceInvalidCharacters:                replaceInvalidCharacters,
+		DefaultDDLCollation:                     defaultDDLCollation,
+		StorageSerializationPolicy:              storageSerializationPolicy,
+		LogLevel:                                logLevel,
+		TraceLevel:                              traceLevel,
+		SuspendTaskAfterNumFailures:             suspendTaskAfterNumFailures,
+		TaskAutoRetryAttempts:                   taskAutoRetryAttempts,
+		UserTaskManagedInitialWarehouseSize:     userTaskManagedInitialWarehouseSize,
+		UserTaskTimeoutMs:                       userTaskTimeoutMs,
+		UserTaskMinimumTriggerIntervalInSeconds: userTaskMinimumTriggerIntervalInSeconds,
+		QuotedIdentifiersIgnoreCase:             quotedIdentifiersIgnoreCase,
+		EnableConsoleOutput:                     enableConsoleOutput,
+		PipeExecutionPaused:                     GetConfigPropertyAsPointerAllowingZeroValue[bool](d, "pipe_execution_paused"),
+		Comment:                                 GetConfigPropertyAsPointerAllowingZeroValue[string](d, "comment"),
 	}
-
-	err := client.Schemas.Create(ctx, sdk.NewDatabaseObjectIdentifier(database, name), createReq)
-	if err != nil {
-		return fmt.Errorf("error creating schema %v err = %w", name, err)
+	if strings.EqualFold(strings.TrimSpace(name), "PUBLIC") {
+		opts.OrReplace = sdk.Pointer(true)
+	}
+	if err := client.Schemas.Create(ctx, id, opts); err != nil {
+		return diag.FromErr(err)
 	}
-
 	d.SetId(helpers.EncodeSnowflakeID(database, name))
 
-	return ReadSchema(d, meta)
+	return ReadContextSchema(ctx, d, meta)
 }
 
-// ReadSchema implements schema.ReadFunc.
-func ReadSchema(d *schema.ResourceData, meta interface{}) error {
+func ReadContextSchema(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	client := meta.(*provider.Context).Client
-	ctx := context.Background()
 	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.DatabaseObjectIdentifier)
 
-	database, err := client.Databases.ShowByID(ctx, id.DatabaseId())
+	_, err := client.Databases.ShowByID(ctx, id.DatabaseId())
 	if err != nil {
 		d.SetId("")
```

Comment:
I'm wondering if we ever need this? And if so maybe it would better to return the error here and not hope that it will be returned when calling show schemas. The worst case would be if we would get error here, but not in the show schemas, then it would be really painful to debug without any log or warning added to Diagnostics.

Reply:
Sometimes sf objects behave poorly when show is invoked and one of the first parts points to a non existing object. This is an interesting problem that could be solved solely on the SDK level for each object, but I think we could leave this discussion for the object renaming (cause it touches the removal topic too).